File: Payload_Type/aphrodite/agent_functions/sleep_cmd.py
from mythic_container.MythicCommandBase import *
from mythic_container.MythicRPC import *
from mythic_container.PayloadBuilder import *


class SleepArguments(TaskArguments):
    def __init__(self, command_line, **kwargs):
        super().__init__(command_line, **kwargs)
        self.args = [
            CommandParameter(
                name="interval",
                type=ParameterType.Number,
                description="Sleep interval in seconds",
                parameter_group_info=[ParameterGroupInfo(group_name="Default", ui_position=0, required=True)],
            ),
            CommandParameter(
                name="jitter",
                type=ParameterType.Number,
                description="Jitter percentage (0-100)",
                default_value=0,
                parameter_group_info=[ParameterGroupInfo(group_name="Default", ui_position=1, required=False)],
            ),
        ]

    async def parse_arguments(self):
        if len(self.command_line) == 0:
            raise ValueError("Interval required")
        parts = self.command_line.strip().split()
        self.add_arg("interval", int(parts[0]))
        if len(parts) > 1:
            self.add_arg("jitter", int(parts[1]))
        else:
            self.add_arg("jitter", 0)


class SleepCommand(CommandBase):
    cmd = "sleep"
    needs_admin = False
    help_cmd = "sleep <seconds> [jitter%]"
    description = "Set callback sleep interval and optional jitter percentage"
    version = 1
    author = "@0xbbuddha"
    argument_class = SleepArguments
    attackmapping = ["T1029"]
    attributes = CommandAttributes(supported_os=[SupportedOS.Linux, SupportedOS.Windows])

    async def create_tasking(self, task: MythicTask) -> MythicTask:
        interval = task.args.get_arg("interval")
        jitter = task.args.get_arg("jitter")
        task.display_params = f"{interval}s jitter={jitter}%"
        return task

    async def process_response(self, response: AgentResponse):
        pass

File: Payload_Type/aphrodite/agent_code/src/commands/sleep_cmd.nim
import std/strutils

proc runSleep*(params: string, sleepInterval: var int, jitterPercent: var int): string =
  ## Update sleep interval and optional jitter.
  ## params format: '{"interval": 30, "jitter": 10}'
  ## or simple: "30" or "30 10"
  let parts = params.strip().split()
  if parts.len == 0 or parts[0].len == 0:
    return "Error: interval required"

  try:
    sleepInterval = parseInt(parts[0])
    if parts.len > 1:
      jitterPercent = parseInt(parts[1])
    result = "Sleep set to " & $sleepInterval & "s with " & $jitterPercent & "% jitter"
  except ValueError:
    result = "Error: invalid interval value"
